ksql-engine/src/test/java/io/confluent/ksql/util/AvroUtilTest.java

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+
+package io.confluent.ksql.util;
+
+import org.apache.kafka.connect.data.Schema;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+
+import io.confluent.kafka.schemaregistry.client.SchemaMetadata;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException;
+import io.confluent.ksql.metastore.KsqlTopic;
+import io.confluent.ksql.metastore.MetaStoreImpl;
+import io.confluent.ksql.parser.KsqlParser;
+import io.confluent.ksql.parser.tree.AbstractStreamCreateStatement;
+import io.confluent.ksql.parser.tree.Statement;
+import io.confluent.ksql.serde.DataSource;
+import io.confluent.ksql.serde.avro.KsqlAvroTopicSerDe;
+
+import static org.easymock.EasyMock.anyObject;
+import static org.easymock.EasyMock.anyString;
+import static org.easymock.EasyMock.expect;
+import static org.easymock.EasyMock.mock;
+import static org.easymock.EasyMock.replay;
+import static org.junit.Assert.assertThat;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.fail;
+
+public class AvroUtilTest {
+
+  String ordersAveroSchemaStr = "{"
+                     + "\"namespace\": \"kql\","
+                     + " \"name\": \"orders\","
+                     + " \"type\": \"record\","
+                     + " \"fields\": ["
+                     + "     {\"name\": \"ordertime\", \"type\": \"long\"},"
+                     + "     {\"name\": \"orderid\",  \"type\": \"long\"},"
+                     + "     {\"name\": \"itemid\", \"type\": \"string\"},"
+                     + "     {\"name\": \"orderunits\", \"type\": \"double\"},"
+                     + "     {\"name\": \"arraycol\", \"type\": {\"type\": \"array\", \"items\": \"double\"}},"
+                     + "     {\"name\": \"mapcol\", \"type\": {\"type\": \"map\", \"values\": \"double\"}}"
+                     + " ]"
+                     + "}";
+
+  @Test
+  public void shouldPassAvroCheck() throws Exception {
+    SchemaRegistryClient schemaRegistryClient = mock(SchemaRegistryClient.class);
+    SchemaMetadata schemaMetadata = mock(SchemaMetadata.class);
```

Comment:
You can just create a `SchemaMetadata` instance rather than mocking it

Reply:
Wouldn't it fail if it cannot connect to the service?

Reply:
`SchemaMetadata` is just a value object. It doesn't connect to any services

Reply:
Wait, yes, somehow I read it as `SchemaRegistryClient`! You're right, will use an instance instead of mock.